### PR TITLE
Allow both bebraven.org and braven.org to iframe stuff.

### PR DESCRIPTION
**Pull request by sadleb**
We have a resume and cover letter bank in the portal that iframe's
a page here on the Join server. We have two portals, one hosted at
booster.braven.org and one at portal.bebraven.org that need to be
able to iframe the resume/cover letter page. Also, soon we will be
migrating portal.bebraven.org to portal.braven.org. This commit
allows those iframes to work.

See: https://app.asana.com/0/1170770467635599/1173810330734585

### Test Plan:
- Load these two pages in prodution and make sure the iframe works:
  https://portal.bebraven.org/courses/1/pages/resume-and-cover-letter-bank
  https://booster.braven.org/courses/1/pages/resume-and-cover-letter-bank
- Make sure it also works in the dev env.
- If you get really ambitious try to iframe it on other domains in production
  and see that it doesn't work.
- If you get really, really ambitious try it out on older browsers and see what happens.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -16,32 +16,30 @@ def prepare_exception_notifier
...
-  # We moved from beyondz.org to join.bebraven.org but I couldn't
-  # get .htaccess redirects working on heroku, so I'm doing it in code.
-  # Don't do it on dev though since you may have setup some funky domain stuff
-  # to develop (e.g. you're using ngrok to expose your server to the internets)
-  def redirect_if_old
-    if Rails.env.production? && request.host != Rails.application.secrets.root_domain
-      redirect_to "#{request.protocol}#{Rails.application.secrets.root_domain}#{request.fullpath}", :status => :moved_permanently
-    end
-  end
+# Note: the move from beyondz.org doesn't need to be supported anymore, but we're
```
Comment (sadleb) [inline app/controllers/application_controller.rb:23]: This is why this is commented out.

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -16,32 +16,30 @@ def prepare_exception_notifier
...
+# about to move from bebraven.org to braven.org and so we'll likely need to bring some
+# variant of this back, so I'm leaving it here commented out.
```
Comment (sadleb) [inline app/controllers/application_controller.rb:25]: This is why this is commented out.

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -16,32 +16,30 @@ def prepare_exception_notifier
...
-    domain = Rails.application.secrets.canvas_server
-    port   = Rails.application.secrets.canvas_port
-    if (secure && port != 443) || (!secure && port != 80)
-      port = ":#{port}"
+    if Rails.env.production?
+      response.headers['Content-Security-Policy'] = "frame-ancestors 'self' *.bebraven.org *.braven.org"
     else
-      port = ''
+      # In dev, don't try to handle all the permutations of how each dev could have their env setup.
+      response.headers['Content-Security-Policy'] = "frame-ancestors *"
```
Comment (sadleb) [inline app/controllers/application_controller.rb:41]: This feels less secure than the old code, but I don't think it's worth doing a bunch of stuff to come up with the exact booster and accelerator URls for both dev and prod. If one of our own websites on our own domains gets hacked and iframes Join pages to do some funky stuff... we have bigger problems.

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -16,32 +16,30 @@ def prepare_exception_notifier
...
     end
-
-    response.headers['X-Frame-Options'] = "ALLOW-FROM http#{secure}://#{domain}#{port}"
```
Comment (sadleb) [inline app/controllers/application_controller.rb:43]: This header is obsolete. I'm fine breaking the iframes on really old browsers.

File: docker-compose/.env-docker [hunk 5]
```diff
@@ -12,6 +12,10 @@ DATABASE_NAME=joindb
 DATABASE_USER=postgres
 DATABASE_PASSWORD=
 
+# Note: the AWS access key and secret must be set in your shell env if you
+# want things that need the bucket to work, like the resume and cover letter bank.
+AWS_BUCKET=beyondz-dev-assets
```
Comment (sadleb) [inline docker-compose/.env-docker:17]: This is where the actual resume/cover letter files are stored in the dev env. I was able to see the page working in dev with this here and using my same AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY for other apps by adding this bucket to the security policy for dev env filestores.

**Comment by sadleb** [comment]
This is too cumbersome to really test properly in dev, so we're going to have to push to staging/prod and iterate if I messed something up.

File: app/controllers/application_controller.rb (hunk 3 above)
Comment (rshipp) [reply app/controllers/application_controller.rb:41]: yeah this is fine imo

**Pull request merged**